examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart

```diff
@@ -116,103 +171,113 @@ class TextFormFieldDemoState extends State<TextFormFieldDemo> {
+                const SizedBox(height: 24.0),
+                new PasswordField(
+                  fieldKey: _passwordFieldKey,
+                  helperText: 'No more than 8 characters.',
+                  labelText: 'Password *',
+                  onSaved: (String value) { person.password = value; },
                 ),
-                maxLines: 1,
-              ),
-              new Row(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: <Widget>[
-                  new Expanded(
-                    child: new TextFormField(
-                      key: _passwordFieldKey,
-                      decoration: const InputDecoration(
-                        hintText: 'How do you log in?',
-                        labelText: 'New Password *',
-                      ),
-                      obscureText: true,
-                      onSaved: (String value) { person.password = value; },
-                    ),
+                const SizedBox(height: 24.0),
+                new TextFormField(
+                  decoration: const InputDecoration(
+                    border: const UnderlineInputBorder(),
+                    filled: true,
+                    labelText: 'Re-type password',
                   ),
-                  const SizedBox(width: 16.0),
-                  new Expanded(
-                    child: new TextFormField(
-                      decoration: const InputDecoration(
-                        hintText: 'How do you log in?',
-                        labelText: 'Re-type Password *',
-                      ),
-                      obscureText: true,
-                      onFieldSubmitted: (String value) { _handleSubmitted(); },
-                      validator: _validatePassword,
-                    ),
+                  maxLength: 8,
+                  onFieldSubmitted: (String value) { person.password = value; },
+                  obscureText: true,
+                  validator: _validatePassword,
+                ),
+                const SizedBox(height: 24.0),
+                new Center(
+                  child: new RaisedButton(
+                    child: const Text('SUBMIT'),
+                    onPressed: _handleSubmitted,
                   ),
-                ],
-              ),
-              new Container(
-                padding: const EdgeInsets.all(20.0),
-                alignment: Alignment.center,
-                child: new RaisedButton(
-                  child: const Text('SUBMIT'),
-                  onPressed: _handleSubmitted,
                 ),
-              ),
-              new Container(
-                padding: const EdgeInsets.only(top: 20.0),
-                child: new Text('* indicates required field', style: Theme.of(context).textTheme.caption),
-              ),
-            ],
+                const SizedBox(height: 24.0),
+                new Text(
+                  '* indicates required field',
```

Comment:
Oh, I see, you put it here.  I think I would just eliminate the "*required" helper texts and just have this one notice, so that all the fields can be consistent.

Reply:
Good point, I've done that.